test-harness/tests/test_agent.py

```diff
@@ -0,0 +1,61 @@
+# Copyright (C) Mesosphere, Inc. See LICENSE file for details.
+
+import copy
+
+import pytest
+
+from generic_test_code import (
+    generic_no_slash_redirect_test,
+    generic_correct_upstream_dest_test,
+    generic_correct_upstream_request_test,
+    generic_upstream_headers_verify_test,
+)
+
+pytestmark = pytest.mark.usefixtures("agent_ar_process")
+
+
+class TestLogsEndpoint():
+    def test_redirect_req_without_slash(self, agent_ar_process):
+        generic_no_slash_redirect_test(agent_ar_process, '/system/v1/logs/v1')
+
+    def test_if_request_is_sent_to_correct_upstream(self,
+                                                    agent_ar_process,
+                                                    superuser_user_header):
+
+        generic_correct_upstream_dest_test(agent_ar_process,
+                                           superuser_user_header,
+                                           '/system/v1/logs/v1/foo/bar',
+                                           'http:///run/dcos/dcos-log.sock',
+                                           )
+
+    @pytest.mark.parametrize("path_given,path_expected",
+                             [("/system/v1/logs/v1/foo/bar", "/foo/bar"),
+                              ("/system/v1/logs/v1/", "/"),
```

Comment:
This is a copy&paste from `test_master.py` which doesn't test `/system/v1/logs/v1` because there is a redirect to `/system/v1/logs/v1/`.